polynote-frontend/polynote/ui/component/notebook/cell.ts

```diff
@@ -415,42 +422,34 @@ class CodeCell extends Cell {
                     });
 
                     this.setErrorMarkers(error, reportInfos)
-
-                    return reportInfos
                 })
             } else {
                 this.clearErrorMarkers("compiler")
-                return []
             }
         })
-        const runtimeErrorState = cellState.mapView<"runtimeError", ErrorEl | undefined>("runtimeError", (runtimeError?: RuntimeError) => {
+
+        const runtimeErrorState = cellState.view("runtimeError")
+        runtimeErrorState.addObserver(runtimeError => {
             if (runtimeError) {
-                const err = ErrorEl.fromServerError(runtimeError.error, this.cellId);
-                const cellLine = err.errorLine;
+                const cellLine = getErrorLine(runtimeError.error, this.cellId);
                 if (cellLine !== undefined && cellLine >= 0) {
                     const model = this.editor.getModel()!;
                     this.setErrorMarkers(runtimeError, [{
-                        message: err.summary.message,
+                        message: runtimeError.error.message, //err.summary.message,
                         startLineNumber: cellLine,
                         endLineNumber: cellLine,
                         startColumn: model.getLineMinColumn(cellLine),
                         endColumn: model.getLineMaxColumn(cellLine),
                         severity: 8
                     }])
                 }
-                return err
             } else {
                 this.clearErrorMarkers("runtime")
-                return undefined
             }
-        }, (errEl1: ErrorEl, errEl2: ErrorEl) => deepEquals(errEl1, errEl2, ["el"])) // remove `el` from equality check because dom elements are never equal.
-        let cellOutput = new CodeCellOutput(dispatcher, cellState, compileErrorsState, runtimeErrorState);
-        this.onDispose.then(() => {
-            compileErrorsState.dispose()
-            runtimeErrorState.dispose()
-            cellOutput.dispose()
         })
 
+        let cellOutput = new CodeCellOutput(dispatcher, cellState, this.cellId, cellState.view("compileErrors"), runtimeErrorState);
```

Comment:
Any reason to make a new `view` here or should this be `compileErrorsState` instead?

Reply:
I guess it should probably be `compileErrorsState`